scripts/src/java/org/oppia/android/scripts/coverage/CoverageRunner.kt

```diff
@@ -0,0 +1,79 @@
+package org.oppia.android.scripts.coverage
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import org.oppia.android.scripts.common.BazelClient
+import org.oppia.android.scripts.common.CommandExecutor
+import org.oppia.android.scripts.common.CommandExecutorImpl
+import org.oppia.android.scripts.common.ScriptBackgroundCoroutineDispatcher
+import java.io.File
+import java.util.concurrent.TimeUnit
+
+/**
+ * Class responsible for running coverage analysis asynchronously.
+ *
+ * @param repoRoot the absolute path to the working root directory
+ * @param targetFile Path to the target file to analyze coverage.
+ */
+class CoverageRunner(
+  private val repoRoot: File,
+  private val scriptBgDispatcher: ScriptBackgroundCoroutineDispatcher
+) {
+
+  /**
+   * Runs coverage analysis asynchronously for the Bazel test target.
+   *
+   * @param repoRoot the absolute path to the working root directory
+   * @param scriptBgDispatcher the [ScriptBackgroundCoroutineDispatcher] to be used for running the coverage command
+   * @param bazelTestTarget Bazel test target to analyze coverage.
+   * @return a deferred value that contains the path of the coverage data file [will contain the proto for the coverage data].
+   */
+  suspend fun runWithCoverageAsync(
+    bazelTestTarget: String
+  ): Deferred<String?> {
+    return CoroutineScope(scriptBgDispatcher).async {
+      val coverageData = getCoverage(bazelTestTarget)
+      val data = coverageData
+      parseCoverageDataFile(data)
+    }
+  }
+
+  /**
+   * Runs coverage command for the Bazel test target.
+   *
+   * @param repoRoot the absolute path to the working root directory
+   * @param scriptBgDispatcher the [ScriptBackgroundCoroutineDispatcher] to be used for running the coverage command
+   * @param bazelTestTarget Bazel test target to analyze coverage.
+   * @return a lisf of string that contains the result of the coverage execution.
+   */
+  fun getCoverage(
+    bazelTestTarget: String
+  ): List<String> {
+    val commandExecutor: CommandExecutor = CommandExecutorImpl(
+      scriptBgDispatcher, processTimeout = 5, processTimeoutUnit = TimeUnit.MINUTES
+    )
+    val bazelClient = BazelClient(repoRoot, commandExecutor)
```

Comment:
Suggest initializing this via a lazy field (like we do in a bunch of other scripts and script utilities).

Reply:
**Done:** Initialized bazelClient using lazy field as done in MavenDependenciesRetriever.kt [[commit](https://github.com/oppia/oppia-android/pull/5423/files#diff-e1015bdd1aafa12e3e95de3b408525733c702992881ece1d2399b57ded6a44fdR23)]